clients/spark/core/src/main/scala/io/treeverse/clients/Exporter.scala

```diff
@@ -130,6 +132,46 @@ object Exporter{
   final val defaultParallelism = 10
 }
 
+object Main {
+  def main(args: Array[String]) {
+    val conf = new Conf(args)
+    println("repo is: " + conf.repo()+", rootLocation is: " + conf.rootLocation())
+
+    val spark = SparkSession.builder().getOrCreate()
+    val sc = spark.sparkContext
+
+    val endpoint = sc.hadoopConfiguration.get("lakefs.api.url")
+    val accessKey = sc.hadoopConfiguration.get("lakefs.api.access_key")
+    val secretKey = sc.hadoopConfiguration.get("lakefs.api.secret_key")
+
+    val apiClient = new ApiClient(endpoint, accessKey, secretKey)
+    val exporter = new Exporter(spark, apiClient, conf.repo(), conf.rootLocation())
+
+    if (conf.commit_id.isSupplied) {
+      exporter.exportAllFromCommit(conf.commit_id())
+      return
+    }
+
+    if (conf.prev_commit_id.isSupplied) {
+      exporter.exportFrom(conf.branch(), conf.prev_commit_id())
+      return
+    }
+
+    exporter.exportAllFromBranch(conf.branch())
+  }
+}
+
+class Conf(arguments: Seq[String]) extends ScallopConf(arguments) {
```

Comment:
It is highly non-trivial how to use this, so documentation of CLI args is probably required here. And thinking of it, given that there are 4 or 5 arguments, mostly optional -- consider making them flags.

Reply:
`repo` and `rootLocation` are mandatory and therefore args.
Everything else is optional, hence the `opt`.

Adding a link to their wiki which is pretty good.

Reply:
Oh gee... the flag name is the variable name.  That's quite surprising.

So much for Scala being a function language (it fails referential transparency, because not only does `opt[String](required=false)` do something different on each call, it even does something different if the name of the variable is changed...).

Anyway, how does the user who just wants to run the program know what args?  Do we get a usage message from scallop if I run with no parameters?